s3/src/main/scala/geotrellis/store/s3/S3LayerDeleter.scala

```diff
@@ -51,17 +52,20 @@ class S3LayerDeleter(
         iter
           .map { s3obj => ObjectIdentifier.builder.key(s3obj.key).build() }
           .toList
-      val deleteDefinition =
-        Delete.builder()
-          .objects(objIdentifiers: _*)
-          .build()
-      val deleteRequest =
-        DeleteObjectsRequest.builder()
-          .bucket(bucket)
-          .delete(deleteDefinition)
-          .build()
-      s3Client.deleteObjects(deleteRequest)
-      attributeStore.delete(id)
+
+      for (objIdentifiersChunk <- objIdentifiers.grouped(requestLimit)) {
```

Comment:
:+1: nice. 

A minor comment that could be just:

```scala
objIdentifiers.grouped(requestLimit).foreach { objIdentifiersChunk =>
  // ...
}
```

But it's up to you to address it or not :D